core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -62,4 +64,12 @@ private[history] abstract class ApplicationHistoryProvider {
    */
   def getConfig(): Map[String, String] = Map()
 
+  /**
+   * Get the [[Path]]s to the Event log files. For legacy event log directories, directory path
+   * itself is returned. The caller is responsible for listing the files and using them as needed.
+   * If the attemptId is [[None]], event logs corresponding to all attempts for the given
+   * application are downloaded as a single zip file.
+   */
+  def getEventLogPaths(appId: String, attemptId: Option[String]): Seq[Path] = Seq.empty
```

Comment:
I thought about this - the idea that I could come up with was to add a new interface and have an implementation that wraps a `Path` and get some of the `Path` functionality from that implementation.